src/core/bonded_interactions/bonded_tab.cpp

```diff
@@ -37,6 +37,11 @@ int tabulated_bonded_set_params(int bond_type,
   make_bond_type_exist(bond_type);
 
   /* set types */
+  if (bonded_ia_params[bond_type].type == BONDED_IA_TABULATED_DISTANCE or
+      bonded_ia_params[bond_type].type == BONDED_IA_TABULATED_ANGLE or
+      bonded_ia_params[bond_type].type == BONDED_IA_TABULATED_DIHEDRAL) {
+    delete bonded_ia_params[bond_type].p.tab.pot;
```

Comment:
cant't you just check for ```nullptr```?

Reply:
``` c++
if (bonded_ia_params[bond_type].p.tab.pot) delete bonded_ia_params[bond_type].p.tab.pot;
```

Reply:
Actually this would raise a Clang-Tidy warning because you can simplify it to
```c++
delete bonded_ia_params[bond_type].p.tab.pot;
```
which is safe on a `nullptr`.
However, this would also trigger a segfault, because you can create a tabulated bond on top of an existing FENE bond, in which case `.p.tab.pot` contains a random address. You can overwrite an bond existing bond in ESPResSo, this is used for example in `testsuite/python/interactions_bonded_interface.py`.